### PR TITLE
[E2E] fix min desired value of MachinePool nodes for spot instance MachinePool test manifest

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
This PR corrects the minimum desired size of spot instances MachinePool, in accordance to https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3253#issuecomment-1050792767

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

File: test/e2e/data/infrastructure-aws/kustomize_sources/machine-pool/spot-instance-machine-pool.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ kind: AWSMachinePool
 metadata:
   name: ${CLUSTER_NAME}-mp-1
 spec:
-  minSize: 1
+  minSize: 0
```
Comment (Skarlso) [inline test/e2e/data/infrastructure-aws/kustomize_sources/machine-pool/spot-instance-machine-pool.yaml:27]: Oh nice. :D :D So this prevented it to scale down to zero? :)

Comment (Ankitasw) [reply test/e2e/data/infrastructure-aws/kustomize_sources/machine-pool/spot-instance-machine-pool.yaml:27]: Yes 😓

**Comment by Skarlso** [comment]
Well done! :)

/lgtm

**Comment by Skarlso** [comment]
/hold
until tests pass

**Comment by Ankitasw** [comment]
/unhold
as tests passed.

**Pull request merged**